### PR TITLE
[MultiDB]:sonic-swss replace old API with new APIs

### PR DESCRIPTION
**Pull request by Pan-XT**
**What I did**
Replace DBConnector old APIs with new APIs
**Why I did it**
Enable Multi DB

I implemented similar changes in https://github.com/sonic-net/sonic-swss/pull/1138

### CODE REVIEW

**Review by eddieruan-alibaba — APPROVED** [review]

**Comment by eddieruan-alibaba** [comment]
Verified in Phoenix wing's daily sanity.

http://phoenixwing.com.cn/vsonic

**Review by eddieruan-alibaba — APPROVED** [review]

**Comment by eddieruan-alibaba** [comment]
Validated with phoenix wing

admin@PE3:~$ ls -l /etc/sonic/enable_multidb 
-rw-r--r-- 1 root root 0 Oct 11 01:46 /etc/sonic/enable_multidb
admin@PE3:~$ ps aux | grep redis
message+     990  0.8  0.5  86764 39228 pts/0    Sl   06:28   7:06 /usr/bin/redis-server 127.0.0.1:6379
message+     991  0.8  0.2  71404 21248 pts/0    Rl   06:28   6:50 /usr/bin/redis-server 127.0.0.1:6380
message+     992  0.3  0.2  64748 17384 pts/0    Sl   06:28   2:28 /usr/bin/redis-server 127.0.0.1:6381
message+     994  2.9  0.3  71404 24000 pts/0    Sl   06:28  23:44 /usr/bin/redis-server 127.0.0.1:6382
message+     996  0.1  0.2  62188 16904 pts/0    Sl   06:28   1:22 /usr/bin/redis-server 127.0.0.1:6383

This PR is a minor fix to update some old api to use new api.

File: fdbsyncd/fdbsyncd.cpp [hunk 1]
```diff
@@ -16,10 +16,10 @@ int main(int argc, char **argv)
 {
     Logger::linkToDbNative("fdbsyncd");
 
-    DBConnector appDb(APPL_DB, DBConnector::DEFAULT_UNIXSOCKET, 0);
+    DBConnector appDb("APPL_DB", 0);
```
Comment (qiluo-msft) [inline fdbsyncd/fdbsyncd.cpp:19]: Just want to double check you are using unix socket. We are moving away from TCP port due to performance and security concerns.

Comment (Pan-XT) [reply fdbsyncd/fdbsyncd.cpp:19]: Yes, I am using a Unix socket.
src/sonic-swss-common/common/dbconnector.cpp lines 674 to 691.

DBConnector::DBConnector(const string& dbName, unsigned int timeout, bool isTcpConn, const SonicDBKey &key)
    : m_dbId(SonicDBConfig::getDbId(dbName, key))
    , m_dbName(dbName)
    , m_key(key)
{
    struct timeval tv = {0, (suseconds_t)timeout * 1000};
    struct timeval *ptv = timeout ? &tv : NULL;
    if (isTcpConn)
    {
        initContext(SonicDBConfig::getDbHostname(dbName, m_key).c_str(), SonicDBConfig::getDbPort(dbName, m_key), ptv);
    }
    else
    {
        initContext(SonicDBConfig::getDbSock(dbName, m_key).c_str(), ptv);
    }

    select(this);
}

Comment (liuh-80) [reply fdbsyncd/fdbsyncd.cpp:19]: The code here actually invoke this ctor, the default value of isTcpConn is false, so it's using unix socket:

    DBConnector(const std::string &dbName, unsigned int timeout, bool isTcpConn = false);

**Review by liuh-80 — APPROVED** [review]

**Pull request merged**